File: sdi3-23.Cli-EJB/src/uo/sdi/client/util/Input.java
package uo.sdi.client.util;

import alb.util.console.Console;

public class Input {

	public static Long pedirLong(String mensaje) {
		Long valor = null;
		do {
			try {
				valor = Console.readLong(mensaje);
			} catch (NullPointerException ne) {
				valor = null;
			}

		} while (valor == null || valor < 0);
		return valor;
	}
	
	
	

}

File: sdi3-23.Cli-SOAPcsharp/sdi3-23.Cli-SOAPcsharp/Input.cs
using alb.util.log;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_23.Cli_SOAPcsharp
{
    class Input
    {
        public static long pedirLong(String mensaje)
        {
            long valor = -1;
            do
            {
                try
                {
                    valor = Consola.readLong(mensaje).longValue();
                }
                catch (NullReferenceException nre)
                {
                    Log.trace("problema a la hora de sacar el long");
                    Log.trace(nre);
                    valor = -1;
                }
            } while (valor < 0);
            return valor;
        }
    }
}
